### PR TITLE
Add LUX blockchain support

### PR DESCRIPTION
**Pull request by giaki3003**
Hello, here's a PR which adds LUX support.
We currently have the explorer running here: https://blockbook.luxcore.io/

### CODE REVIEW

File: bchain/coins/lux/luxrpc.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+	}
+	s.RPCMarshaler = btc.JSONMarshalerV1{}
+	s.ChainConfig.SupportsEstimateSmartFee = true
+
+ 	return s, nil
+}
+
+ // Initialize initializes LuxRPC instance.
+func (b *LuxRPC) Initialize() error {
+	chainName, err := b.GetChainInfoAndInitializeMempool(b)
```
Comment (martinboehm) [inline bchain/coins/lux/luxrpc.go]: The method `GetChainInfoAndInitializeMempool` was removed, please use here the following code instead:

```
	ci, err := b.GetChainInfo()
	if err != nil {
		return err
	}
	chainName := ci.Chain
```

**Comment by martinboehm** [comment]
Hi, thank you for your contribution!

I have a couple of comments:
1) we have today merged a feature branch, which slightly affects your code in `luxrpc.go`. Please rebase your code and update your `luxrpc.go` code as I commented in the code review.
2) could you use ports xx55 instead of xx69? They will fit better to the port number series.
3) add integration tests

**Comment by giaki3003** [comment]
> Hi, thank you for your contribution!
> 
> I have a couple of comments:
> 
> 1. we have today merged a feature branch, which slightly affects your code in `luxrpc.go`. Please rebase your code and update your `luxrpc.go` code as I commented in the code review.
> 2. could you use ports xx55 instead of xx69? They will fit better to the port number series.
> 3. add integration tests

Thanks for getting back at me! Will rebase + fix the parser file, and will also add some integration tests (maybe some examples could help). But i'd love to keep the port number if possible. Let me know

File: bchain/coins/lux/luxrpc.go (hunk 1 above)
Comment (giaki3003) [reply bchain/coins/lux/luxrpc.go]: Thanks!

**Comment by martinboehm** [comment]
1) please rebase your changes in such a way that only your modifications are in the changeset.
2) OK, keep the port :)
3) you can check for example [this pull request](https://github.com/trezor/blockbook/pull/133/files) to see some integration tests

**Comment by giaki3003** [comment]
> 1. please rebase your changes in such a way that only your modifications are in the changeset.
> 2. OK, keep the port :)
> 3. you can check for example [this pull request](https://github.com/trezor/blockbook/pull/133/files) to see some integration tests

Thank you for the suggestion. Will rack up some data for the integration tests

**Comment by martinboehm** [comment]
Closed due to inactivity.

**Pull request closed without merging**